data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaSourceSchemaConsumer.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.consumer;
+
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.kafka.configuration.KafkaSourceConfig;
+import org.opensearch.dataprepper.plugins.kafka.configuration.TopicsConfig;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * * An interface with a generic method which helps to process the records for
+ * avro or json or plain text consumer dynamically.
+ */
+@SuppressWarnings("deprecation")
+public interface KafkaSourceSchemaConsumer<K, V> {
```

Comment:
minor: this can renamed to something else, like drop Schema and make it KafkaSourceConsumer.

Reply:
Resolved. Renamed the class to KafkaSourceConsumer